src/js/utils/translation-helper.ts

```diff
@@ -0,0 +1,79 @@
+import { En } from '../lang/en.js';
+import { Es } from '../lang/es.js';
+import { Fr } from '../lang/fr.js';
+import { Pt } from '../lang/pt.js';
```

Comment:
I'll have a closer look later but a blocker here is that all the languages are bundled up which will make the bundle size increase too much. 

en.js is the default and has to be included but the rest should be separately importable via 
`media-chrome/lang/xx.js` for example

they should also be able to provide their own lang file with a simple method.

```js
import { addTranslation } from 'media-chrome';

export const Pt: TranslationType = {
  tooltips: {
    enterAirplay: 'Iniciar AirPlay',
  },
  nouns: {
    audioPlayer: 'reprodutor de áudio',
  },
  verbs: {
    play: 'reproduzir',
  },
};

addTranslation(Pt);
```

Reply:
yes, for media-chrome's own translation files we can call that `addTranslation()` call in the file.
the imports or the script tags (for HTML only) have to come before the Media Chrome custom elements markup.
that should make it possible I think right.

@ronalduQualabs make sure the lookup of the translated string is done at time of accessing the string in the UI component. Not during the file execution like it is now for `tooltipLabels` for example. It needs to be a lazy getter for this to work.

Reply:
> have to come before the Media Chrome custom elements markup. that should make it possible I think right
Could be. Any issues using `<script type="module">` for either in this case? Could you write out how to see this working in the HTML use case?

Reply:
true the order has to be correct then, that's a bit of a bummer.

```html
<script type="module" src="https://cdn.jsdelivr.net/npm/media-chrome@4/lang/nl.js/+esm"></script>
<script type="module" src="https://cdn.jsdelivr.net/npm/media-chrome@4/+esm"></script>

<media-controller>...</media-controller>
```

nl.js

```js
import { addTranslation } from 'media-chrome/utils/translation.js';

addTranslation({
...
});
```

Reply:
Ok so then how does addTranslation work if media-chrome itself isn't in the page yet?

Reply:
it's just a normal esm module (separate import) with a dictionary object that is shared.
first we populate the dictionary, later we access it when the main bundle is run.

Reply:
@luwes Maybe I'm not completely understanding this, because wouldn't this cause a race condition? 

<img width="1774" alt="Screenshot 2025-03-25 at 17 56 46" src="https://github.com/user-attachments/assets/5b06f5dd-026c-4ea4-996c-dbccd735646a" />

see: https://codepen.io/davidvanleeuwen/pen/raNrzqj

And most linters would turn around the order, so this is quite a frustrating setup now.

ps: [chrome allows you to change locale](https://www.aleksandrhovhannisyan.com/blog/changing-locale-in-chrome-with-dev-tools/)